src/cluster.rs

```diff
@@ -14,79 +21,110 @@ use std::{
 use tokio_sync::{mpsc, oneshot, watch};
 use tokio_timer::Interval;
 
-#[derive(Debug, Default, Clone)]
-pub struct Event;
+pub struct Cluster<T, Target>
+where
+    T: Transport<Target>,
```

Comment:
Yeah, I'm not sure yet if I want to keep this or have it be an associated type. I'm gonna punt on that for now.

Reply:
Definitely associated type. Since the address the cluster is listening on is associated to it. Probably have a `ToSocketAddr` bound on it.

Reply:
Yeah, I also need to flesh out Endpoint so that its based on bytes. Then we can have a bound `Into<Endpoint>` for target. Then impl From<SocketAddr> for Endpoint etc

Reply:
I agree, that would make for a cleaner API.